src/app/cli/src/commands/system_diagnose_command.rs

```diff
@@ -0,0 +1,197 @@
+        verification_svc: Arc<dyn VerificationService>,
+    ) -> Result<(), CLIError> {
+        let diagnose_check = RunCheck::new();
+        diagnose_check.run(dataset_repo, verification_svc).await
+    }
+}
+
+pub struct RunCheck {
+    container_runtime: ContainerRuntime,
+}
+
+impl RunCheck {
+    fn new() -> Self {
+        Self {
+            container_runtime: ContainerRuntime::default(),
+        }
+    }
+
+    async fn run_container_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "container installed... ")?;
+
+        let is_container_installed = match self
+            .container_runtime
+            .custom_cmd("--version".to_string())
+            .output()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_installed}")?;
+        Ok(())
+    }
+
+    async fn run_container_pull_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer can pull images... ")?;
+
+        let is_container_image_pulled =
+            match self.container_runtime.pull_image(DUMMY_IMAGE, None).await {
+                Ok(_) => style("ok").green(),
+                Err(_) => style("failed").red(),
+            };
+
+        write!(f, "{is_container_image_pulled}")?;
+        Ok(())
+    }
+
+    async fn run_container_rootless_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer rootless check... ")?;
+
+        let is_container_rootless = match self
+            .container_runtime
+            .run_attached(DUMMY_IMAGE)
+            .init(true)
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_rootless}")?;
+        Ok(())
+    }
+
+    async fn run_container_volume_mount_check(&self, f: &mut Stdout) -> Result<(), CLIError> {
+        write!(f, "\ncontainer volume mounts work... ")?;
+        let temp_dir = tempfile::tempdir()?;
+        let cwd = Path::new(".").canonicalize()?;
+
+        let file_path = temp_dir.path().join("tmp.txt");
+        let _ = File::create(file_path)?;
+
+        let is_container_rootless = match self
+            .container_runtime
+            .run_attached(DUMMY_IMAGE)
+            .volume((cwd, temp_dir.into_path()))
+            .init(true)
+            .spawn()
+            .unwrap()
+            .wait()
+            .await
+        {
+            Ok(_) => style("ok").green(),
+            Err(_) => style("failed").red(),
+        };
+
+        write!(f, "{is_container_rootless}")?;
+        Ok(())
+    }
+
+    async fn run_workspace_consistency_check(
+        &self,
+        dataset_repo: Arc<dyn DatasetRepository>,
+        verification_svc: Arc<dyn VerificationService>,
+        f: &mut Stdout,
```

Comment:
Maybe we should use a more generic type, i.e. something that supports `std::io::Write` trait ? Who knows, if we would like to use some other type of I/O stream in future.